include/openssl/core.h

```diff
@@ -185,15 +185,15 @@ typedef void (*OSSL_thread_stop_handler_fn)(void *arg);
  *              provider needs it.  This value is passed to other provider
  *              functions, notably other context constructors.
  */
-typedef int (OSSL_provider_init_fn)(const OSSL_CORE_HANDLE *handle,
+typedef int (OSSL_FUNC_provider_init_fn)(const OSSL_CORE_HANDLE *handle,
```

Comment:
Absolutely not!  The provider init function isn't a dispatched function, it's a module/provider entry point.

Reply:
Agreed. Is this the only exceptional case which needs to be reverted?

Reply:
[Reverted](https://github.com/openssl/openssl/pull/12222/commits/ae887295945c9e1aea224f8176c321653dc629b6).